integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkApplicationExecutionContext.java

```diff
@@ -144,14 +152,21 @@ private OpenLineage.ParentRunFacet buildApplicationParentFacet() {
   }
 
   private OpenLineage.JobBuilder getJobBuilder() {
-    String name =
-        eventEmitter
-            .getOverriddenAppName()
-            .orElse(olContext.getSparkContext().map(SparkContext::appName).orElse("unknown"));
-    return openLineage
-        .newJobBuilder()
-        .namespace(eventEmitter.getJobNamespace())
-        .name(normalizeName(name));
+    return openLineage.newJobBuilder().namespace(eventEmitter.getJobNamespace()).name(getJobName());
+  }
+
+  private String getJobName() {
+    return jobNameProviders.stream()
+        .map(provider -> provider.getJobName(olContext))
```

Comment:
wouldn't it make sense to add extra method to an interface verifying if a job provider is applicable. Something like `isDefinedAt(OpenLineageContext context)`. Such an approach is common within this codebase.

Reply:
I was thinking about it. It requires checking the same parameters while checking if the provider is applicable and when it provides the value. But it should never be a performance bottleneck so I refactored the code.